packages/app/src/options.js

```diff
@@ -1,46 +1,87 @@
     hideLegend: { defaultValue: false, requestable: false },
     noSpaceBetweenColumns: { defaultValue: false, requestable: false },
     percentStackedValues: { defaultValue: false, requestable: false },
-    rangeAxisDecimals: { defaultValue: null, requestable: false },
-    rangeAxisLabel: { defaultValue: null, requestable: false },
-    rangeAxisMaxValue: { defaultValue: null, requestable: false },
-    rangeAxisMinValue: { defaultValue: null, requestable: false },
-    rangeAxisSteps: { defaultValue: null, requestable: false },
+    rangeAxisDecimals: { defaultValue: 0, requestable: false },
+    rangeAxisLabel: { defaultValue: '', requestable: false },
+    rangeAxisMaxValue: { defaultValue: 0, requestable: false },
+    rangeAxisMinValue: { defaultValue: 0, requestable: false },
+    rangeAxisSteps: { defaultValue: 0, requestable: false },
     regressionType: { defaultValue: 'NONE', requestable: false },
     showData: { defaultValue: true, requestable: false },
-    targetLineLabel: { defaultValue: null, requestable: false },
-    targetLineValue: { defaultValue: null, requestable: false },
+    targetLineLabel: { defaultValue: '', requestable: false },
+    targetLineValue: { defaultValue: 0, requestable: false },
     // legendDisplayStrategy
     // legendSet
-    aggregationType: { defaultValue: null, requestable: true },
+    aggregationType: { defaultValue: '', requestable: true },
     completedOnly: { defaultValue: false, requestable: true },
     hideSubtitle: { defaultValue: false, requestable: false },
     hideTitle: { defaultValue: false, requestable: false },
     sortOrder: { defaultValue: 0, requestable: false },
-    subtitle: { defaultValue: null, requestable: false },
-    title: { defaultValue: null, requestable: false },
+    subtitle: { defaultValue: '', requestable: false },
+    title: { defaultValue: '', requestable: false },
     // topLimit
 };
 
+export const computedOptions = {
+    baseLine: { defaultValue: false, requestable: false },
+    targetLine: { defaultValue: false, requestable: false },
+};
+
 export default options;
 
-export const getOptionsDefaultValues = () => {
-    return Object.entries(options).reduce((map, [option, props]) => {
-        map[option] = props.defaultValue;
+export const getOptionsForUi = () => {
+    return Object.entries({ ...options, ...computedOptions }).reduce(
+        (map, [option, props]) => {
+            map[option] = props.defaultValue;
 
-        return map;
-    }, {});
+            return map;
+        },
+        {}
+    );
 };
 
 export const getOptionsForRequest = () => {
     return Object.entries(options).filter(
         ([option, props]) => props.requestable
     );
 };
+
+export const getOptionsFromVisualization = visualization => {
+    const optionsFromVisualization = {
+        ...getOptionsForUi(),
+        ...getPropsByKeys(visualization, Object.keys(options)),
+    };
+
+    if (
```

Comment:
To improve the readability of this code, what about something like this? (not tested)

```
const isNotDefault = prop => {
    return (
        optionsFromVisualization[prop] &&
        optionsFromVisualization[prop] !== options[prop].defaultValue
    );
};

optionsFromVisualization.baseLine =
    isNotDefault('baseLineLabel') || isNotDefault('baseLineValue');

optionsFromVisualization.targetLine =
    isNotDefault('targetLineLabel') || isNotDefault('targetLineValue');
```